scripts: tools/docker/run.sh: fix usage docker network name passed via cmdline arg

MuteSpirit opened the pull request:
Script "tools/docker/run.sh" assumes that gateway/repeater container
will be attached to Docker network named "prplMesh-net-${UNIQUE_ID}" and
ignore network name passed via cmdline argument.

For real boards usage purposes it's need to be able pass custom Docker
network name to provide connectivity of gateway container with choosen
board.

Fix assignement network name with value from args if the last is exist.
Fix checking the presense of created container. Current solution does
not correctly match, for example, "test" name if
"test-and-the-rest-name" is already present.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: tools/docker/run.sh: fix usage docker network name passed via cmdline arg

Script "tools/docker/run.sh" assumes that gateway/repeater container
will be attached to Docker network named "prplMesh-net-${UNIQUE_ID}" and
ignore network name passed via cmdline argument.

For real boards usage purposes it's need to be able pass custom Docker
network name to provide connectivity of gateway container with choosen
board.

Fix assignement network name with value from args if the last is exist.
Fix checking the presense of created container. Current solution does
not correctly match, for example, "test" name if
"test-and-the-rest-name" is already present.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

vitalybu commented on tools/docker/run.sh:
```diff
@@ -80,19 +80,19 @@ main() {
     }
 
-    NETWORK=prplMesh-net-${UNIQUE_ID}
+    NETWORK=${NETWORK:-prplMesh-net-${UNIQUE_ID}}
```
```suggestion
    NETWORK="${NETWORK:-prplMesh-net-${UNIQUE_ID}}"
```

MuteSpirit replied:
Fixed

vitalybu commented on tools/docker/run.sh:
```diff
@@ -80,19 +80,19 @@ main() {
     [ -z "$IPADDR" -a -n "$NETWORK" ] && {
         dbg "Generate random IP for container $NAME for network $NETWORK"
-        IPADDR=$(generate_container_random_ip $NETWORK)
+        IPADDR=$(generate_container_random_ip ${NETWORK})
```
shellcheck SC2086:
```suggestion
        IPADDR="$(generate_container_random_ip "${NETWORK}")"
```

MuteSpirit replied:
fixed

vitalybu reviewed (approved):
Approving as my only comments are shellcheck related nitpicks.
But it would be great if you could add another commit fixing all shellcheck issues.

tomereli reviewed (approved):
I removed the ready for merge since it looks like there are shellcheck issues - so please fix those before you add the label back. Otherwise LGTM.

Commit:
scripts: fix shellcheck issues

tests/test_gw_repeater.sh:
  remove quoting "run.sh" arguments  which
  already defined like PARAM="$2"
  Reason: "./tests/test_gw_repeater.sh" call "run.sh" with additinal
  two empty string agruments and broke docker entrypoint work

tools/docker/run.sh: fix shellcheck warnings:
  use double quoting to avoid spit arguments
  `...` -> $(...)
  $(($var)) -> $((var))

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>
